feature_engine/timeseries/forecasting/expanding_window_features.py

```diff
@@ -0,0 +1,232 @@
+# Author: Kishan Manani
+# License: BSD 3 clause
+from __future__ import annotations
+
+from typing import List, Optional
+
+import pandas as pd
+from sklearn.utils.validation import check_is_fitted
+from statsmodels.tsa.tsatools import freq_to_period
+
+from feature_engine.docstrings import (
+    Substitution,
+    _drop_original_docstring,
+    _feature_names_in_docstring,
+    _fit_not_learn_docstring,
+    _fit_transform_docstring,
+    _missing_values_docstring,
+    _n_features_in_docstring,
+    _variables_numerical_docstring,
+)
+from feature_engine.timeseries.forecasting.base_forecast import BaseForecast
+
+
+@Substitution(
+    variables=_variables_numerical_docstring,
+    missing_values=_missing_values_docstring,
+    drop_original=_drop_original_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class ExpandingWindowFeatures(BaseForecast):
+    """
+    ExpandingWindowFeatures adds new features to a dataframe based on window
+    operations. Expanding window operations are operations that perform an
+    aggregation over an expanding window of all past values relative to the
+    value of interest. An expanding window feature is, in other words, a feature
+    created after computing statistics (e.g., mean, min, max, etc.) using a
+    window over all the past data. For example, the mean value of all months
+    prior to the month of the interest is an expanding window feature.
+
+    ExpandingWindowFeatures uses the pandas' functions `expanding()`, `agg` and
```

Comment:
I give all these detail in WindowFeatures, because we are using parameters and inputs for all these pandas methods, and the users need to define them. So basically, I am trying (not sure I am succeeding) to make users see the link between WindoFeatures and the methods from pandas, so that, if they know these methods already, using WindowFeatures will be straighforward. And if they don't know them, they can go and check them out.

I mention this because I don't think this class is exposing the methods for pandas shift. Is this the case? If we are not exposing params for this method, I would try and reduce the documentation to the necessary bits to undertand how to use this class.

Reply:
Even if the parameters for `shift` are not exposed, the `functions` parameter that go into the `agg` is still exposed and is tightly coupled to methods in Pandas. So I think there is still a good reason for this documentation to be here if the objective is to "see the link between WindoFeatures and the methods from pandas, so that, if they know these methods already, using WindowFeatures will be straighforward".